drivers/net/bnxt/bnxt_ethdev.c

```diff
@@ -1134,45 +1133,34 @@ static int bnxt_set_vf_vlan_filter_op(struct rte_eth_dev *dev, uint16_t vlan,
 			if (vlan_on) {
 				/* First, search for a duplicate... */
 				for (j=0; j<bp->pf.vf_info[i].vlan_count; j++) {
-					if (bp->pf.vf_info[i].vlan_table[j].vid == vlan)
-						continue;
-				}
-				if (j < bp->pf.vf_info[i].vlan_count)
-					continue;
-
-				/* Now check that there's space */
-				if (bp->pf.vf_info[i].vlan_count == getpagesize() / sizeof(struct bnxt_vlan_table_entry)) {
-					RTE_LOG(ERR, PMD, "VF %d VLAN table is full, cannot add VLAN %u\n", i, vlan);
-					rc = -1;
-					continue;
+					if (rte_be_to_cpu_16(bp->pf.vf_info[i].vlan_table[j].vid) == vlan)
+						break;
 				}
+				if (j == bp->pf.vf_info[i].vlan_count) {
+					/* Now check that there's space */
+					if (bp->pf.vf_info[i].vlan_count == getpagesize() / sizeof(struct bnxt_vlan_table_entry)) {
+						RTE_LOG(ERR, PMD, "VF %d VLAN table is full, cannot add VLAN %u\n", i, vlan);
+						rc = -1;
+						continue;
+					}
 
-				/* And finally, add to the end of the table */
-				ve = &bp->pf.vf_info[i].vlan_table[bp->pf.vf_info[i].vlan_count++];
-				ve->tpid = rte_cpu_to_be_16(0x8100);	// TODO: Hardcoded TPID
-				ve->vid = vlan;
+					/* And finally, add to the end of the table */
+					ve = &bp->pf.vf_info[i].vlan_table[bp->pf.vf_info[i].vlan_count++];
+					ve->tpid = rte_cpu_to_be_16(0x8100);	// TODO: Hardcoded TPID
+					ve->vid = rte_cpu_to_be_16(vlan);
+				}
 			}
 			else {
 				for (j=0; j<bp->pf.vf_info[i].vlan_count; j++) {
-					if (bp->pf.vf_info[i].vlan_table[j].vid != vlan)
+					if (rte_be_to_cpu_16(bp->pf.vf_info[i].vlan_table[j].vid) != vlan)
 						continue;
 					memmove(&bp->pf.vf_info[i].vlan_table[j], &bp->pf.vf_info[i].vlan_table[j+1],
 					    getpagesize() - ((j+1) * sizeof(struct bnxt_vlan_table_entry)));
 					j--;
 					bp->pf.vf_info[i].vlan_count--;
 				}
 			}
-			dflt_vnic = bnxt_hwrm_func_qcfg_vf_dflt_vnic_id(bp, i);
-			if (dflt_vnic < 0) {
-				// This simply indicates there's no driver loaded.  This is not an error.
-				RTE_LOG(ERR, PMD, "Unable to get default VNIC for VF %d\n", i);
-			}
-			else {
-				if (bnxt_hwrm_vnic_qcfg(bp, &vnic, dflt_vnic) == 0) {
-					if (bnxt_hwrm_cfa_l2_set_rx_mask(bp, &vnic, bp->pf.vf_info[i].vlan_count, bp->pf.vf_info[i].vlan_table))
-						rc = -1;
-				}
-			}
+			rte_pmd_bnxt_set_vf_vlan_anti_spoof(dev->data->port_id, i, bp->pf.vf_info[i].vlan_spoof_en);
```

Comment:
Should we not call bnxt_hwrm_cfa_l2_set_rx_mask(bp, &vnic,
bp->pf.vf_info[i].vlan_count,
bp->pf.vf_info[i].vlan_table) after this rte_pmd_bnxt_set_vf_vlan_anti_spoof()?
We are updating the vlan_table in driver, but only calling vlan_anti_spoof,
and not issuing the HWRM command to update vlans in hardware.